src/connector/src/source/filesystem/opendal_source/opendal_enumerator.rs

```diff
@@ -51,14 +52,22 @@ impl<Src: OpendalSource> SplitEnumerator for OpendalEnumerator<Src> {
 
     async fn list_splits(&mut self) -> ConnectorResult<Vec<OpendalFsSplit<Src>>> {
         let empty_split: OpendalFsSplit<Src> = OpendalFsSplit::empty_split();
+        let prefix = self.prefix.as_deref().unwrap_or("/");
 
-        Ok(vec![empty_split])
+        match self.op.list(prefix).await {
+            Ok(_) => return Ok(vec![empty_split]),
+            Err(e) => {
+                return Err(anyhow!(e)
+                    .context("fail to create source, please check your config.")
```

Comment:
maybe support with more info, such as source_id, or the source can be hard to find in catalogs.

Reply:
Here we just want to let users know about some configuration errors on the frontend, and the specific error code (such as `PermissionDenied`) will be printed, like
```
Caused by these errors (recent errors listed first):
  1: gRPC request to meta service failed: Internal error
  2: failed to create source worker
  3: fail to create source, please check your config.
  4: PermissionDenied (persistent) at List::next, context: { uri: http://127.0.0.1:9301/hummock001?delimiter=%2F&list-type=2, response: Parts { status: 403, version: HTTP/1.1, headers: {"connection": "close", "content-length": "407", "accept-ranges": "bytes", "content-type": "application/xml", "date": "Tue, 20 Aug 2024 02:53:10 GMT", "server": "MinIO", "strict-transport-security": "max-age=31536000; includeSubDomains", "vary": "Origin", "vary": "Accept-Encoding", "x-amz-id-2": "5e0466557790194f71acb9d44341f26b381bbf7ca42536c367b0a3b121df5c06", "x-amz-request-id": "17ED502DE1BFD178", "x-content-type-options": "nosniff", "x-xss-protection": "1; mode=block"} }, service: s3, path: /, listed: 0 } => S3Error { code: "SignatureDoesNotMatch", message: "The request signature we calculated does not match the signature you provided. Check your key and signing method.", resource: "/hummock001", request_id: "17ED502DE1BFD178" }
```